src/main/java/com/ust/sourcecourse/configuration/service/ProjectGroupService.java

```diff
@@ -62,6 +62,9 @@ private ProjectGroupResponse getProjectGroupresponse(ProjectGroup projectgrp) {
 
 	public List<ProjectGroupResponse> findByProjectUid(Long uid) {
 		List<ProjectGroup> projectGroups = projectGroupRepository.findByProjectUid(uid);
+		if (projectGroups.isEmpty()) {
+			throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Project groups not found for project UID: " + uid);
```

Comment:
If it's empty, no need to return with 404-NOT_FOUND. Instead return with 201 - NO_CONTENT